api/proto/teleport/legacy/types/types.proto

```diff
@@ -5132,6 +5132,67 @@ message KubernetesResourceSpecV1 {
   string Namespace = 1 [(gogoproto.jsontag) = "namespace"];
 }
 
+// MaintenanceWindowV1 is a config singleton used to determine the appropirate window
+// for performing infrequent periodics that have the potential to degrade service.
+message MaintenanceWindowV1 {
+  ResourceHeader Header = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "",
+    (gogoproto.embed) = true
+  ];
+  MaintenanceWindowSpecV1 Spec = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "spec"
+  ];
+  // Nonce is used to protect against concurrent modification of the maintenance
+  // window. Clients should treat nonces as opaque.
+  uint64 Nonce = 3 [(gogoproto.jsontag) = "nonce,omitempty"];
+}
+
+// MaintenanceWindowSpecV1 encodes the parameters of the upgrade window config object.
+message MaintenanceWindowSpecV1 {
+  // AgentUpgrades encodes the agent upgrade window.
+  AgentUpgradeWindow AgentUpgrades = 1 [(gogoproto.jsontag) = "agent_upgrades,omitempty"];
+}
+
+// AgentUpgradeWindow is the config object used to determine upcoming agent
+// upgrade windows.
+message AgentUpgradeWindow {
+  // UTCStartHour is the start hour of the maintenance window in UTC.
+  uint32 UTCStartHour = 1 [(gogoproto.jsontag) = "utc_start_hour"];
+  // Weekdays is an optional list of weekdays. If not specified, an agent upgrade window
+  // occurs every day.
+  repeated string Weekdays = 2 [(gogoproto.jsontag) = "weekdays,omitempty"];
+}
+
+// ScheduledAgentUpgradeWindow is a dervied value representing a single
+// upgrade window. Generated by the AgentUpgradeWindow config object.
+message ScheduledAgentUpgradeWindow {
```

Comment:
I don't quite understand the comment here. What's the difference between `AgentUpgradeWindow` and `ScheduledAgentUpgradeWindow`?